integration/helpers.go

```diff
@@ -184,3 +184,22 @@ func CollectKeys(t testing.TB, doc bson.D) []string {
 
 	return res
 }
+
+// AssertEqualWriteError compares expected mongo.WriteError Message and Code with actual error.
+func AssertEqualWriteError(t *testing.T, expected *mongo.WriteError, actual error) bool {
+	t.Helper()
+
+	writeException, ok := actual.(mongo.WriteException)
+	if !ok {
+		return assert.Equal(t, expected, actual)
+	}
+
+	if len(writeException.WriteErrors) != 1 {
+		return assert.Equal(t, expected, actual)
+	}
+
+	actualWriteErr := writeException.WriteErrors[0]
+
+	return assert.Equal(t, expected.Message, actualWriteErr.Message) &&
```

Comment:
`assert.Equal` would print message text or code if it's not equal. The potential problem here is that if the code and message are not equal it will print a message or a code not both of them, so you would require 2 test runs to fix it.